servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/HeaderUtils.java

```diff
@@ -185,7 +186,9 @@ private static void updateResponseContentLength(final int contentLength, final H
                 for (int i = 0; i < items.size(); i++) {
                     contentLength += calculateContentLength(items.get(i));
                 }
-                flatRequest = Publisher.<Object>from(metadata).concat(fromIterable(items));
+                flatRequest = Publisher.<Object>from(metadata)
+                        .concat(fromIterable(items))
+                        .concat(succeeded(EmptyHttpHeaders.INSTANCE));
```

Comment:
There was a path in the original approach, where the codec was part of the payload API, that the ObjectEncoder was hanging due to missing trailers. It might have been due to filtering-out the trailers during encoding. I will remove for now, and unless I can reproduce.

Reply:
Removed. The actual reason this was hanging was the attempt to calculate content length on non-streaming requests. A check for content-encoding was missing to skip this length-calculation and use chunking, thus this logic was anticipating a single buffer and not multiple chunks.